pkg/proxy/provider/http/http.go

```diff
@@ -0,0 +1,164 @@
+package http
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/cenkalti/backoff/v4"
+	"github.com/containous/traefik/v2/pkg/config/dynamic"
+	"github.com/containous/traefik/v2/pkg/job"
+	"github.com/containous/traefik/v2/pkg/log"
+	"github.com/containous/traefik/v2/pkg/provider"
+	"github.com/containous/traefik/v2/pkg/safe"
+	"github.com/containous/traefik/v2/pkg/types"
+)
+
+var _ provider.Provider = (*Provider)(nil)
+
+// Provider is a provider.Provider implementation that queries an endpoint for a configuration.
+type Provider struct {
+	Endpoint     string         `description:"Load configuration from this endpoint." json:"endpoint" toml:"endpoint" yaml:"endpoint" export:"true"`
+	PollInterval types.Duration `description:"Polling interval for endpoint." json:"pollInterval,omitempty" toml:"pollInterval,omitempty" yaml:"pollInterval,omitempty"`
+	PollTimeout  types.Duration `description:"Polling timeout for endpoint." json:"pollTimeout,omitempty" toml:"pollTimeout,omitempty" yaml:"pollTimeout,omitempty"`
+}
+
+// New creates a new instance of the HTTP provider.
+func New(endpoint string, pollInterval types.Duration, pollTimeout types.Duration) *Provider {
```

Comment:
This provider is used by the traefik core, and I'm not sure this would work with that logging method would it? We don't have a fieldlogger in the proxy node.

Reply:
Ah! true!